File: src/procedures/SimpleBPMProcedure.java
package procedures;

import java.awt.Color;
import java.awt.geom.Area;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;

import led.LEDDataBundle;
import led.LEDStripManager;
import led.ProcedureBundleFields;

public class SimpleBPMProcedure extends Procedure {

	public ArrayList<Area> mAreas = new ArrayList<Area>();
	
	public int mBpm = 60;
	public float mBeatStep = 0.0f;
	float mCounter = 0.0f;
	
	public SimpleBPMProcedure(LEDDataBundle _bundle) {
		super(_bundle);
		
		mBeatStep = (60.0f / (float)mBpm)*1000.0f;
		System.out.println("BEATSTEP: " + mBeatStep);
	}

	@Override
	public void start() {
		
	}
	
	@Override
	void update() {
		mCounter = mCounter > mBeatStep ? 0.0f : mCounter + 20.0f;
		
		mStrip.setArea(0, LEDStripManager.LED_COUNT, Color.BLACK);
		if (mCounter == 0.0f) {
			DateTimeFormatter dtf = DateTimeFormatter.ofPattern("mm:ss:SSS");  
			LocalDateTime now = LocalDateTime.now();  
			System.out.println(dtf.format(now));  
			   
			mStrip.setArea(0, LEDStripManager.LED_COUNT, Color.RED);
		}
	}
}

File: src/procedures/SimpleBPMProcedure.kt
package procedures

import led.LEDDataBundle
import procedures.Procedure
import led.ColorRGB
import led.ColorRGBA
import interpolation.InterpolationType
import led.LEDStripManager
import interpolation.Interpolation
import javax.script.ScriptEngine
import javax.script.Invocable
import java.lang.NoSuchMethodException
import javax.script.ScriptException
import javax.script.ScriptEngineManager
import led.json.LEDJsonProcedure
import java.nio.file.Files
import java.nio.file.Paths
import led.json.interpreter.LEDInterpreter
import led.json.interpreter.FileVersions
import java.io.IOException
import procedures.models.navigation.PositionMarker
import procedures.ProcedureCalls
import procedures.ProcedureType
import procedures.BootCompleteProcedure
import procedures.ColorInstantSetProcedure
import procedures.FadeInFadeOutProcedure
import procedures.FadeToMultiColorProcedure
import procedures.FadeToUniformColorProcedure
import procedures.FillStripProcedure
import procedures.FillStripInterpolatedProcedure
import procedures.RainbowProcedure
import procedures.RainbowMonoProcedure
import procedures.SimpleBPMProcedure
import procedures.MultiProcedure
import procedures.BlinkProcedure
import procedures.GlitterProcedure
import procedures.models.IndeterminateState
import led.ColorHSV
import java.awt.Color
import java.awt.geom.Area
import java.time.format.DateTimeFormatter
import java.time.LocalDateTime
import java.util.ArrayList

class SimpleBPMProcedure(_bundle: LEDDataBundle) : Procedure(_bundle) {
    var mAreas = ArrayList<Area>()
    var mBpm = 60
    var mBeatStep = 0.0f
    var mCounter = 0.0f
    override fun start() {}
    public override fun update() {
        mCounter = if (mCounter > mBeatStep) 0.0f else mCounter + 20.0f
        mStrip!!.setArea(0, LEDStripManager.LED_COUNT, Color.BLACK)
        if (mCounter == 0.0f) {
            val dtf = DateTimeFormatter.ofPattern("mm:ss:SSS")
            val now = LocalDateTime.now()
            println(dtf.format(now))
            mStrip!!.setArea(0, LEDStripManager.LED_COUNT, Color.RED)
        }
    }

    init {
        mBeatStep = 60.0f / mBpm.toFloat() * 1000.0f
        println("BEATSTEP: $mBeatStep")
    }
}